### PR TITLE
Revised Ninja Logic

### PR DESCRIPTION
**Pull request by jlsnow301**
## About The Pull Request
This change does two things:
Updates ninja antag check to the proper M?.mind?.x
Also reduces ninja objectives to 3 because they were getting more than possible, ie, they would get an empty objective on spawn

## Why It's Good For The Game
bug fixes.

## Changelog
:cl:
tweak: is_ninja proc
tweak: Ninja objectives shouldn't give empty objectives now
/:cl:

### CODE REVIEW

File: code/modules/antagonists/ninja/ninja.dm [hunk 1]
```diff
@@ -105,7 +105,7 @@
 	return TRUE
 
 /proc/is_ninja(mob/living/M)
-	return M && M.mind && M.mind.has_antag_datum(/datum/antagonist/ninja)
+	return M?.mind?.has_antag_datum(/datum/antagonist/ninja)
```
Comment (ike709) [inline code/modules/antagonists/ninja/ninja.dm:108]: Are you sure this actually fixes the empty objective bug? `M && M.mind && M.mind.has_antag_datum(/datum/antagonist/ninja)` and `M?.mind?.has_antag_datum(/datum/antagonist/ninja)` are just two different ways of writing the same code. Unless BYOND is being stupid and returning `M` for some reason.

Comment (ike709) [reply code/modules/antagonists/ninja/ninja.dm:108]: Oh nevermind I'm a smoothbrain. The quantity was the objective issue.

**Review by ike709 — APPROVED** [review]
I still think ninja needs a complete rewrite and should be disabled pending that, but this is good enough for now.

**Comment by jlsnow301** [comment]
@ike709 I agree, they're only good on paper really, could use a massive rework.

**Review by MarkSuckerberg — APPROVED** [review]
Seems good enough to fix it for now.

**Pull request merged**